### PR TITLE
Add RockDB Subcommand for printing usage per column family

### PR DESCRIPTION
**Pull request by siladu**
Pretty much a lift and shift of @ahamlat's https://github.com/ahamlat/RocksdDB-Column-Families-Size

```
➜  besu git:(rocksdb-usage-subcommand) $BESU storage x-rocksdb usage --help
Usage: besu storage x-rocksdb usage [-hV]
Prints disk usage
  -h, --help      Show this help message and exit.
  -V, --version   Print version information and exit.
  
➜  besu git:(rocksdb-usage-subcommand) $BESU storage x-rocksdb --help
Usage: besu storage x-rocksdb [-hV] [COMMAND]
Print RocksDB information
  -h, --help      Show this help message and exit.
  -V, --version   Print version information and exit.
Commands:
  usage  Prints disk usage
```

```
$BESU --data-path=/tmp/besu storage x-rocksdb usage
****** Column family 'BLOCKCHAIN' size: 7 KiB ******
Number of keys : 70
Number of live snapshots : 0
Total size of SST Files : 4 KiB
Size of live SST Filess : 4 KiB
Column family size (with getColumnFamilyMetaData) : 4 KiB

****** Column family 'ACCOUNT_INFO_STATE' size: 1 KiB ******
Number of keys : 2
Number of live snapshots : 0
Total size of SST Files : 1 KiB
Size of live SST Filess : 1 KiB
Column family size (with getColumnFamilyMetaData) : 1 KiB

****** Column family 'TRIE_BRANCH_STORAGE' size: 1 KiB ******
Number of keys : 7
Number of live snapshots : 0
Total size of SST Files : 2 KiB
Size of live SST Filess : 2 KiB
Column family size (with getColumnFamilyMetaData) : 2 KiB

****** Column family 'TRIE_LOG_STORAGE' size: 5 KiB ******
Number of keys : 103
Number of live snapshots : 0
Total size of SST Files : 7 KiB
Size of live SST Filess : 7 KiB
Column family size (with getColumnFamilyMetaData) : 7 KiB

****** Column family 'VARIABLES' size: 1 KiB ******
Number of keys : 6
Number of live snapshots : 0
Total size of SST Files : 2 KiB
Size of live SST Filess : 2 KiB
Column family size (with getColumnFamilyMetaData) : 2 KiB
```

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbSubCommand.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import org.rocksdb.Options;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.ParentCommand;
+
+/** The RocksDB subcommand. */
+@Command(
+    name = "x-rocksdb",
```
Comment (jframe) [inline besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbSubCommand.java:36]: Is this intended to be a hidden subcommand with x prefix? Don't think it needs to be hidden or have the x prefix.

File: besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbUsageHelper.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+/** RocksDB Usage subcommand helper methods for formatting and printing. */
+public class RocksDbUsageHelper {
+
+  static void printUsageForColumnFamily(
+      final RocksDB rocksdb, final ColumnFamilyHandle cfHandle, final PrintWriter out)
+      throws RocksDBException {
+    String size = rocksdb.getProperty(cfHandle, "rocksdb.estimate-live-data-size");
+    boolean emptyColumnFamily = false;
+    if (!size.isEmpty() && !size.isBlank()) {
+      long sizeLong = Long.parseLong(size);
```
Comment (jframe) [inline besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbUsageHelper.java:37]: I find the output of this difficult to read with multiple sizes for the column family. A single value for the size in table format ideally a markdown table would be much easier to read.

e.g.
```
| Column Family      | Size  | Keys |
| -------------------| ----- | -----|
| BLOCKCHAIN         | 7 KiB |   70 |
| ACCOUNT_INFO_STATE | 1 KiB |    2 |
```

**Comment by gfukushima** [comment]
I've changed it to follow the MD format suggested by @jframe. Also have removed the experimental prefix.

File: besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbSubCommand.java (hunk 1 above)
Comment (gfukushima) [reply besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbSubCommand.java]: removed the prefix

File: besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbUsageHelper.java (hunk 2 above)
Comment (gfukushima) [reply besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbUsageHelper.java]: changed it to be a md table

File: crypto/algorithms/src/main/java/org/hyperledger/besu/crypto/KeyPairUtil.java [hunk 3]
```diff
@@ -84,6 +84,7 @@ public static KeyPair loadKeyPair(final File keyFile) {
     final KeyPair key;
     if (keyFile.exists()) {
 
+      LOG.info("Attempting to load public key from {}", keyFile.getAbsolutePath());
```
Comment (jframe) [inline crypto/algorithms/src/main/java/org/hyperledger/besu/crypto/KeyPairUtil.java:87]: Was this change intentional? Doesn't look related to the subcommand

File: besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbUsageHelper.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      final RocksDB rocksdb, final ColumnFamilyHandle cfHandle, final PrintWriter out)
+      throws RocksDBException {
+    final String size = rocksdb.getProperty(cfHandle, "rocksdb.estimate-live-data-size");
+    boolean emptyColumnFamily = false;
+    if (!size.isEmpty() && !size.isBlank()) {
+      final long sizeLong = Long.parseLong(size);
+      if (sizeLong == 0) {
+        emptyColumnFamily = true;
+      }
+      String totolSstFilesSize = rocksdb.getProperty(cfHandle, "rocksdb.total-sst-files-size");
```
Comment (jframe) [inline besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbUsageHelper.java:40]: rename totolSstFilesSize to totalSstFilesSize

File: besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbUsageHelper.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  static void printLine(
+      final PrintWriter out,
+      final String cfName,
+      final String keys,
+      final String columnSize,
+      final String sstFilesSize) {
+    final String format = "| %-30s | %-15s | %-12s | %-15s |\n";
+    out.format(format, cfName, keys, columnSize, sstFilesSize);
+  }
+}
```
Comment (garyschulte) [inline besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbUsageHelper.java:95]: using https://github.com/vdmeer/asciitable this could be replaced by:
```
AsciiTable at = new AsciiTable();
at.addRow("Column Family", "Keys", "Column Size", "SST FIles Size")
at.addRule()
```
per cf: `at.addRow(cfName, keys, columnSize, sstFilesSize);`

and finally `out.print(at.render());`

File: besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbUsageHelper.java (hunk 4 above)
Comment (gfukushima) [reply besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/RocksDbUsageHelper.java]: done

**Review by fab-10 — APPROVED** [review]
